modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -481,6 +493,38 @@ object ElasticRequest {
     }
   }
 
+  sealed trait UpdateByQueryRequest
+      extends ElasticRequest[UpdateByQueryResult]
+      with HasRefresh[UpdateByQueryRequest]
+      with HasRouting[UpdateByQueryRequest] {
+    def conflicts(value: UpdateConflicts): UpdateByQueryRequest
+  }
+
+  private[elasticsearch] final case class UpdateByQuery(
+    index: IndexName,
+    script: Script,
+    conflicts: Option[UpdateConflicts],
+    query: Option[ElasticQuery[_]],
+    refresh: Option[Boolean],
+    routing: Option[Routing]
+  ) extends UpdateByQueryRequest { self =>
+    def conflicts(value: UpdateConflicts): UpdateByQueryRequest =
+      self.copy(conflicts = Some(value))
+
+    def refresh(value: Boolean): UpdateByQueryRequest =
+      self.copy(refresh = Some(value))
+
+    def routing(value: Routing): UpdateByQueryRequest =
+      self.copy(routing = Some(value))
+
+    def toJson: Json = {
+      val queryToJson: Json  = query.fold(Obj())(q => q.toJson)
```

Comment:
Can you use `_.toJson`?

Reply:
Even better `query.foldLeft(Obj("script" -> script.toJson))((json, q) => json merge q.toJson)`. 

Something like this. 😄